package/libs/openssl/Makefile

```diff
@@ -156,27 +300,22 @@ ifndef CONFIG_OPENSSL_WITH_SRP
   OPENSSL_OPTIONS += no-srp
 endif
 
-ifeq ($(CONFIG_OPENSSL_OPTIMIZE_SPEED),y)
-  TARGET_CFLAGS := $(filter-out -Os,$(TARGET_CFLAGS)) -O3
+ifndef CONFIG_OPENSSL_WITH_ASM
+  OPENSSL_OPTIONS += no-asm
 endif
 
-ifeq ($(CONFIG_x86_64),y)
-  OPENSSL_TARGET:=linux-x86_64-openwrt
-  OPENSSL_MAKEFLAGS += LIBDIR=lib
-else
-  OPENSSL_OPTIONS+=no-sse2
-  ifeq ($(CONFIG_mips)$(CONFIG_mipsel),y)
-    OPENSSL_TARGET:=linux-mips-openwrt
-  else ifeq ($(CONFIG_aarch64),y)
-    OPENSSL_TARGET:=linux-aarch64-openwrt
-  else ifeq ($(CONFIG_arm)$(CONFIG_armeb),y)
-    OPENSSL_TARGET:=linux-armv4-openwrt
-  else
-    OPENSSL_TARGET:=linux-generic-openwrt
-    OPENSSL_OPTIONS+=no-perlasm
+ifdef CONFIG_i386
+  ifndef CONFIG_OPENSSL_SSE2
+    OPENSSL_OPTIONS += no-sse2
   endif
 endif
 
+OPENSSL_TARGET:=linux-$(CONFIG_ARCH)-openwrt
+
+ifeq ($(CONFIG_x86_64)$(CONFIG_mips64)$(CONFIG_mips64el),y)
+  OPENSSL_MAKEFLAGS += LIBDIR=lib
```

Comment:
There is another way to configure the `LIBDIR=lib` thing

```
./Configure --prefix=/usr --openssldir=/etc/ssl --libdir=lib
```

ref:

https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/openssl#n33
https://github.com/openssl/openssl/blob/OpenSSL_1_1_1/Configure#L787
https://github.com/openssl/openssl/blob/OpenSSL_1_1_0i/Configure#L675

https://github.com/openssl/openssl/blob/OpenSSL_1_0_2p/Configure#L953 which becomes https://github.com/openssl/openssl/blob/OpenSSL_1_0_2p/Configure#L1729